content/docs/details/techdocs/index.md

```diff
@@ -0,0 +1,248 @@
+- [mkdocs-minify-plugin](https://github.com/byrnereese/mkdocs-minify-plugin) - minify HTML, JS or CSS files
+- [mkdocs-git-revision-date-localized-plugin](https://github.com/timvink/mkdocs-git-revision-date-localized-plugin) - enables displaying the date of the last git modification of a page
+- [mkdocs-glightbox](https://github.com/blueswen/mkdocs-glightbox) - display images, iframes, inline content and videos
+- [markdown-inline-mermaid](https://github.com/mermaid-js/mermaid) - generate diagrams from markdown-like text.
+
+
+## Theme and Styling
+Backstage uses an opinionated theme based on [material-mkdocs](https://github.com/backstage/mkdocs-techdocs-core#theme).
+
+NB: Some styles will always be overridden regardless of the mkdocs-material plugin theme settings and this can cause unexpected behavior for those who override the theme setting in a mkdocs.yaml file.
+
+
+## Testing and previewing your documentation
+
+You can generate / serve your docs locally to view what they would look like when they are deployed to Roadie.
+
+To generate the docs to the site directory of the project you can run the following command:
+
+```bash
+npx @techdocs/cli generate --docker-image roadiehq/techdocs
+```
+
+To start a local server at port 3000 containing the generated docs, you can run the following command:
+```bash
+npx @techdocs/cli serve --docker-image roadiehq/techdocs
+```
+
+NB: We have seen some issues generating and serving plantuml and mermaid diagrams sometimes on M1 Macbooks due to unresolved
+bugs in open source dependencies. Please reach out to us anyway if you run into any difficulties. 
+
+## Navigation and Titles
+
+By default, the structure of the docs pages will mirror that of the file system. You can also explicitly describe your
+page structure using the `nav` object in your mkdocs.yaml. Both approaches are described [here](https://www.mkdocs.org/user-guide/writing-your-docs/#file-layout).
+
+Similarly, mkdocs will determine a title for your document according to [these rules](https://www.mkdocs.org/user-guide/writing-your-docs/#meta-data).
+
+## Graphs and Diagrams
+
+### Using Mermaid Diagrams
+
+Roadie supports using Mermaid JS to render diagrams. It must first be enabled in your `mkdocs.yaml`:
+
+```yaml
+markdown_extensions:
+  - markdown_inline_mermaid
+```
+
+Then you can add mermaid diagrams as follows:
+````
+```mermaid
+%%{init: {'theme': 'forest'}}%%
+graph LR
+  A[Start] --> B{Error?};
+  B -->|Yes| C[Hmm...];
+  C --> D[Debug];
+  D --> B;
+  B ---->|No| E[Yay!];
+```
+````
+
+### Customizing graph styling
+
+In order to customize the look of the graphs you will need to use the [Graphviz attributes](https://graphviz.org/doc/info/attrs.html). Setting different values for specific set of attributes will result in graph being rendered that way. For example, let's say we want to change background color from white to lightblue in following graph:
```

Comment:
This might need a preface that we support Graphviz or something. It kinda starts abruptly just describing tech details immediately